### DISCUSSION TITLE
How to apply RandScaleCrop on batches of 3D images?

### DISCUSSION THREAD

**Discussion by codingS3b** [Project-MONAI/MONAI#3583] [General]
I recently ran into the problem of trying to apply the `monai.transforms.RandScaleCrop` to a batch of images which failed.
Below is the code I tried:

```py
from monai import transforms as MT
import torch

transforms = MT.Compose([
    MT.RandScaleCrop(
        roi_scale=(.05, .05, .05),  # minimum crop size
        max_roi_scale=(.05, .8, .8),
        random_center=False,
        random_size=True),
    # no matter what the original crops shape was, we want a fixed size as output
    # at least in the y and x dimension
    MT.Resize(spatial_size=(-1, 64, 64))
])

# monai works on single 3d images -> OK
print(transforms(torch.ones(1, 34, 32, 32)).shape)

# this fails
print(transforms(torch.ones(1, 1, 34, 32, 32)).shape)
```
Is there a way that I'm missing on how to apply the transformation on a batch of images or is this usage not supported?

Thanks for helping out here!

**Comment by wyli** [comment]
Hi @codingS3b, good question! most of the transforms in MONAI currently assumes the 'channel-first' format. Batch input is not directly supported, but it's possible to have:
```py
from monai.data import decollate_batch

outputs = transforms(decollate_batch(torch.ones(2, 1, 34, 32, 32)))
for idx, p in enumerate(outputs):
    print(idx, p.shape)
```
where `decollate_batch` will make the mini-batch a list of channel-first image patches, and `Compose` (by default with `map_item=True`) will apply the transforms to each of the image patches:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/compose.py#L108

**Reply by codingS3b** [reply]
Thanks a lot for your explanation @wyli!
